modules/vpc/main.tf

```
@@ -20,24 +20,47 @@ resource "aws_subnet" "private_subnets" {
 
 ## Public Subnets
 resource "aws_subnet" "public_subnets" {
-  count             = length(var.config.aws_azs)
-  vpc_id            = aws_vpc.vpc.id
-  cidr_block        = var.config.public_subnets[count.index]
-  availability_zone = var.config.aws_azs[count.index]
+  count                   = length(var.config.aws_azs)
+  vpc_id                  = aws_vpc.vpc.id
+  cidr_block              = var.config.public_subnets[count.index]
+  availability_zone       = var.config.aws_azs[count.index]
+  map_public_ip_on_launch = true
 
   tags = {
     Name = format("%s-pub-%s", var.vpc_name, count.index)
   }
 }
 
-## Internet Gateway
+## Elastic IP
+resource "aws_eip" "ngw-eip" {
```

Just a little thing, can be please stick with underscores for resources names? it is weird to mix hyphens and underscores when referencing a resource like `aws_eip.ngw-eip.id`

```suggestion
resource "aws_eip" "ngw_eip" {
```